File: Ansh Pahwa/Day2/PlusOne.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
    int temp=0,n=digits.size()-1;

        if(digits[n]+1>9)
        {
         
            for(int i=digits.size()-1;i>=0;i--)
            {

                if(digits[i]==9 )
                {
                    
                    temp++;
                }
                if(digits[i]+1>9 )
                {
                    
                    digits[i]=0;
                }

                else if(digits[i]+1>9 && digits[i-1]!=9){
                    digits[i]=0;
                    temp=0;
                }

          
                else if(digits[i]<9)
                {
                    digits[i]=digits[i]+1;
                    break;
                }
            }
        
                if(temp==n+1){
                digits.push_back(0);
                digits[0]=1;
                }
           
        }


      
   
        else
           digits[n]=digits[n]+1;
        
    
      
      
    return digits;
    }
};

File: Members/Ashish Singh/Day 2/PlusOne.java
class Solution {
    public int[] plusOne(int[] digits) {

        int n = digits.length;
        for(int i=n-1; i>=0; i--) {
            if(digits[i] < 9) {
                digits[i]++;
                return digits;
            }
            // if digits[i]==9 make corresponding value 0
            digits[i] = 0;
        }
        // handle the case where all digits are 9
        int[] newNumber = new int [n+1];
        newNumber[0] = 1;

        return newNumber;
    }
}